cs.c

```diff
@@ -546,9 +548,11 @@ size_t CAPSTONE_API cs_disasm(csh ud, const uint8_t *buffer, size_t size, uint64
 			mci.flat_insn->size = insn_size;
 
 			// map internal instruction opcode to public insn ID
-			handle->insn_id(handle, insn_cache, mci.Opcode);
 
-			handle->printer(&mci, &ss, handle->printer_info);
+			handle->insn_id(handle, insn_cache, mci.Opcode);
+#ifndef CAPSTONE_DIET
+  			handle->printer(&mci, &ss, handle->printer_info);
```

Comment:
the printer may produce the details for the cs_insn struct. i am wondering how you use Capstone: without detail info, or with detail info? if you do, dont you miss anything?